src/main/scala/fastq.scala

```diff
@@ -92,43 +188,47 @@ case object fastq {
       rec(raw, Seq.empty, false)
     }
 
-    val toPhred33: Int => Char = i => (i + 33).toChar
+    val toPhred33: Int => Char = { i => (i + 33).toChar }
   }
 
-  // TODO value should not have @ at the beginning
-  class Id(val value: String) extends AnyVal {
+  class Id private[fastarious] (val value: String) extends AnyVal {
 
     def asString: String =
       s"@${value}"
   }
+
   case object Id {
 
-    private val isValid: String => Boolean =
-      raw =>
-        (raw startsWith "@") &&
-        !(raw contains "\n")
+    private val isValid: String => Boolean = { raw =>
+      (raw startsWith "@") &&
+      !(raw contains "\n")
+    }
 
     def from(raw: String): Option[Id] = {
 
       // just *one* line
+      // FIXME: raw starts with several @, it will drop them all
       val l = raw.filterNot(_ == '\n').dropWhile(_ == '@')
 
       if(l.nonEmpty) Some( new Id(l) ) else None
     }
 
+    // FIXME: this passes the raw string (with @) to the constructor
     def parseFrom(raw: String): Option[Id] =
       if(isValid(raw)) Some( new Id(raw) ) else None
```

Comment:
I'm confused by these two methods:  `from` and `parseFrom`. The latter just checks the string, but then doesn't drop `@`..